packages/db-objects-redis/lib/objects/objectsInRedisClient.js

```diff
@@ -4479,6 +4456,75 @@ class ObjectsInRedisClient {
         return this.client.get(`${this.metaNamespace}objects.protocolVersion`);
     }
 
+    /**
+     * Sets current host as primary if no primary host missing
+     * Value will expire after sec seconds
+     *
+     * {number} ms - ms until value expires
+     * @return {Promise<number>}
+     */
+    async setPrimaryHost(ms) {
+        if (!this.client) {
+            throw new Error(utils.ERRORS.ERROR_DB_CLOSED);
+        }
+
+        // try to acquire lock
+        const lockAquired = await this.client.evalsha([
+            this.scripts['redlock_acquire'],
+            3,
+            `${this.metaNamespace}objects.primaryHost`,
+            this.settings.hostname,
+            ms
+        ]);
+
+        // we have not acquired lock, maybe we are already master so we should try to extend
+        if (!lockAquired) {
```

Comment:
Should we not remember if we are master (or not) to optimize the calls here a bit? Else anyone basically will always execute both scripts, but in fact:

* master should try to enhance and only starover if not succesful
* "non-masters" should try to get the master and only if "empty" try to get it ... or?

Reply:
But yes, will basically work that way ... :-)

Reply:
Yes, if we introduce a variable to save that we are primary, but for this we need (as discussed) to subscribe for e.g. `evicted` events in objects db too. 

Then if we are primary and get evicted on the primary host state, clean the state (if we are able to register it because if we have lost host, then something is wrong with us).

In all cases if we register `evicted` for the prim host state -> try to set us as primary

Reply:
But then if we trust our cache, in the interval we would never try to acquire only acquire if primaryHostLost triggered, in interval do nothing if we are not primary. if we are primary extend the lock. Other persepctives?

Reply:
or do a "get" in the interval on a "non master" ... and just on the master do extend ... if results are "different then expected" then do full "try to get it"